File: tools/lxdoc/tests/fixtures/sample_function.js
function add(a, b) {
    return a + b;
}

function multiply(x, y) {
    return x * y;
}

class Calculator {
    constructor() {
        this.history = [];
    }

    compute(op, a, b) {
        let result;
        if (op === "add") {
            result = add(a, b);
        } else if (op === "mul") {
            result = multiply(a, b);
        } else {
            throw new Error(`Unknown operation: ${op}`);
        }
        this.history.push({ op, a, b, result });
        return result;
    }
}

module.exports = { add, multiply, Calculator };

File: tools/lxdoc/tests/fixtures/sample_function.py
def add(a, b):
    return a + b


def multiply(x, y):
    return x * y


class Calculator:
    def __init__(self):
        self.history = []

    def compute(self, op, a, b):
        if op == "add":
            result = add(a, b)
        elif op == "mul":
            result = multiply(a, b)
        else:
            raise ValueError(f"Unknown operation: {op}")
        self.history.append((op, a, b, result))
        return result
